src/hooks/useAppState/reducer.js

```diff
@@ -27,24 +34,45 @@ const reducer = (state, { type, data }) => {
         key,
         text,
       };
-      return { ...state, items: [...state.items, newItem] };
+      return {
+        ...state,
+        items: state.items.map((items, index) => {
+          if (index !== state.index) {
+            return items;
+          }
+          return [...items, newItem];
+        }),
+      };
```

Comment:
Adding a new item to a list is a bit more complicated now since it's a two dimensional list